### PR TITLE
split sigstore TUF client into separate package

### PR DESCRIPTION
**Pull request by bdehamer**
#### Summary
Adds a new `@sigstore/tuf` package to house the Sigstore-specific TUF client. Currently, this code lives in the main `sigstore` package. Once this package has been published we'll remove the duplicate code in `sigstore` and integrate with the new package for all TUF-related functions.

### CODE REVIEW

File: packages/tuf/src/__tests__/target.test.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+          TUFError
+        );
+      });
+
+      it('throws an error with code TUF_DOWNLOAD_TARGET_ERROR', async () => {
+        expect.assertions(1);
+        await readTarget(tuf, targetPath).catch((err) => {
+          expect(err.code).toEqual('TUF_DOWNLOAD_TARGET_ERROR');
+        });
+      });
```
Comment (bdehamer) [inline packages/tuf/src/__tests__/target.test.ts:90]: Now I want to move my `toThrowWithCode` helper into a shared package so I can re-use it here 😆

File: packages/tuf/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+```
+
+## Usage
+
+```javascript
+const { initTUF } = require('sigstore')
+```
+
+```javascript
+import { initTUF } from 'sigstore'
```
Comment (feelepxyz) [inline packages/tuf/README.md:30]: Require the tuf package?
```suggestion
const { initTUF } = require('@sigstore/tuf')
```

```javascript
import { initTUF } from '@sigstore/tuf'
```

File: packages/tuf/README.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+## Installation
+
+```
+npm install @sigstore/tuf
+```
+
+## Usage
+
+```javascript
+const { initTUF } = require('sigstore')
```
Comment (feelepxyz) [inline packages/tuf/README.md:26]:
```suggestion
const { initTUF } = require('@sigstore/tuf')
```

File: packages/tuf/README.md (hunk 2 above)
Comment (feelepxyz) [inline packages/tuf/README.md:30]:
```suggestion
import { initTUF } from '@sigstore/tuf'
```

File: packages/tuf/README.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+```
+
+```javascript
+import { initTUF } from 'sigstore'
+```
+
+### initTUF([options])
+Returns a TUF client which can be used to retrieve any target from the Sigstore
```
Comment (feelepxyz) [inline packages/tuf/README.md:34]: Should we also add some readme for the client interface like `getTarget`?

**Review by feelepxyz — APPROVED** [review]

File: packages/tuf/README.md (hunk 3 above)
Comment (bdehamer) [reply packages/tuf/README.md]: updated

File: packages/tuf/README.md (hunk 2 above)
Comment (bdehamer) [reply packages/tuf/README.md]: updated

**Review by feelepxyz — APPROVED** [review]

**Review by feelepxyz — APPROVED** [review]

**Pull request merged**